### PR TITLE
Add the strings task solution

### PR DESCRIPTION
**Pull request by TheVitik**

### CODE REVIEW

File: src/strings/Strings.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+class Strings implements StringsInterface
+{
+
+    /**
+     * The $input variable contains text in snake case (i.e. hello_world or this_is_home_task)
+     * Transform it into camel cased string and return (i.e. helloWorld or thisIsHomeTask)
+     * @see http://xahlee.info/comp/camelCase_vs_snake_case.html
+     *
+     * Make sure the next PHPDoc instructions will be added:
+     * @param string $input
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php]: Please don't use annotations for `string`

File: src/strings/Strings.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     * @return string
+     */
+    public function snakeCaseToCamelCase(string $input): string
+    {
+        $input = strtolower($input);
+        if(!str_contains($input,"_")){
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php]: Please refactor it according to the [PSR-12](https://www.php-fig.org/psr/psr-12/)

File: src/strings/Strings.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     * Transform it into camel cased string and return (i.e. helloWorld or thisIsHomeTask)
+     * @see http://xahlee.info/comp/camelCase_vs_snake_case.html
+     *
+     * Make sure the next PHPDoc instructions will be added:
+     * @param string $input
+     * @return string
+     */
+    public function snakeCaseToCamelCase(string $input): string
+    {
+        $input = strtolower($input);
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php]: Why are you using `strtolower` here? Snake case cannot contain uppercase letters.

File: src/strings/Strings.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    /**
+     * The $input variable contains multibyte text like 'ФЫВА олдж'
+     * Mirror each word individually and return transformed text (i.e. 'АВЫФ ждло')
+     * !!! do not change words order
+     *
+     * Make sure the next PHPDoc instructions will be added:
+     * @param string $input
+     * @return string
+     */
+    public function mirrorMultibyteString(string $input): string
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php:22]: Please try to simplify it.
In my opinion, there should only be 3 steps.
1. explode
2. reverse
3. implode

File: src/strings/Strings.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+     * @param string $noun
+     * @return string
+     */
+    public function getBrandName(string $noun): string
+    {
+        $noun = strtolower($noun);
+        if($noun[0]===$noun[-1]){
+            return ucfirst(substr($noun,0,-1).$noun);
+        }
+        return "The ".ucfirst($noun);
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php]: Please use single quotes here

File: src/strings/Strings.php [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+     * Make sure the next PHPDoc instructions will be added:
+     * @param string $noun
+     * @return string
+     */
+    public function getBrandName(string $noun): string
+    {
+        $noun = strtolower($noun);
+        if($noun[0]===$noun[-1]){
+            return ucfirst(substr($noun,0,-1).$noun);
+        }
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php:38]: Please add an empty line before `return`

File: src/strings/Strings.php [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        return "The ".ucfirst($noun);
+    }
+}
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php]: Please add an empty line in the end of the file

**Review by oleksandr-chornovol — CHANGES_REQUESTED** [review]
Please fix all comments

File: src/strings/Strings.php (hunk 4 above)
Comment (TheVitik) [reply src/strings/Strings.php:22]: image
Then the result on the screenshot

Comment (oleksandr-chornovol) [reply src/strings/Strings.php:22]: It is because you reversed the whole array.
Instead you should use `strrev` for each array item

Comment (TheVitik) [reply src/strings/Strings.php:22]: But I have an error with strrev, because there is a problem with encoding. So I decided to do more steps.

Comment (oleksandr-chornovol) [reply src/strings/Strings.php:22]: OK. Let it be. Just fix the quotes here

File: src/strings/Strings.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace strings;
+
+class Strings implements StringsInterface
+{
+    public function snakeCaseToCamelCase(string $input): string
+    {
+        if (!str_contains($input, '_')) {
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php]: Please add a space after `!`

File: src/strings/Strings.php [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+            return $input;
+        }
```
Comment (oleksandr-chornovol) [inline src/strings/Strings.php:11]: Please add an empty line after the `if` condition

**Review by oleksandr-chornovol — APPROVED** [review]
Good job!
Just some small comments about code style

**Pull request merged**